engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.cpp

```diff
@@ -129,7 +129,11 @@ ral::execution::task_result DistributeAggregateKernel::do_process(std::vector< s
     // num_partitions = context->getTotalNodes() will do for now, but may want a function to determine this in the future.
     // If we do partition into something other than the number of nodes, then we have to use part_ids and change up more of the logic
     int num_partitions = this->context->getTotalNodes();
-    
+
+    // we want to update the `columns_to_hash` because the input could have more columns after ComputeAggregateKernel
+    std::tie(columns_to_hash, std::ignore, std::ignore, std::ignore) = ral::operators::modGroupByParametersPostComputeAggregations(group_column_indices,
```

Comment:
please move this to where you originally get these parameters. WHere you call `parseGroupByExpression` so that you only call it once and so that the group by parameters to be used by the kernel are these modified ones

Reply:
`parseGroupByExpression` is never used in the `DistributionAggregationKernel`.

Reply:
done!